src/client.ts

```diff
@@ -242,14 +265,44 @@ export class Client {
   /**
    * Translates a client route like `/foo/bar` to the full URL used in the host
    * app for the same page, e.g. `https://hostapp.com/#/client-app/foo/bar`.
-   * You should use this whenver generating an internal link within a client
+   * You should use this whenever generating an internal link within a client
    * application so that the user gets a nice experience if they open a link in
    * a new tab, or copy and paste a link URL into a chat message or email.
    *
    * @param clientRoute The /-separated path within the client app to link to.
    */
-  public asHostUrl(clientRoute: string): string {
-    const trimedClientRoute = stripLeadingSlashAndHashTag(clientRoute);
+  public urlFromClientPath(clientRoute: string): string {
+    const hostRootUrl = this.environmentData.hostRootUrl;
+    const assignedRoute = this._assignedRoute || '';
+    const trimmedClientRoute = stripLeadingSlashAndHashTag(clientRoute);
+    return joinRoutes(hostRootUrl, assignedRoute, trimmedClientRoute);
+  }
+
+  /**
+   * Translates a host route like `/app2` to the full URL used in the host
+   * app, e.g. `https://hostapp.com/#/app2`.
+   * You should use this whenever generating a host link within a client
+   * application so that the user gets a nice experience if they open a link in
+   * a new tab, or copy and paste a link URL into a chat message or email.
+   *
+   * @param hostRoute The /-separated path within the host app to link to.
+   */
+  public urlFromHostPath(hostRoute: string): string {
+    const hostRootUrl = this.environmentData.hostRootUrl;
+    const trimmedHostRoute = stripLeadingSlashAndHashTag(hostRoute);
+    return joinRoutes(hostRootUrl, trimmedHostRoute);
+  }
+
+  /**
+   * Deprecated -- replaced with urlFromClientPath
+   * Translates a client route like `/foo/bar` to the full URL used in the host
+   * app for the same page, e.g. `https://hostapp.com/#/client-app/foo/bar`.
+   *
+   * @param clientRouteLegacy Deprecated. The /-separated path within the client app to link to.
```

Comment:
We should add/use the [tsdoc @deprecated flag](https://tsdoc.org/pages/tags/deprecated/) here instead of the descriptions.